mantidproject/mantid
# Fix CTWA and generate workspace

***Note: PR #21417 should be merged before this PR***. Until that is merged it might be easier to look at the [comparison with that branch](https://github.com/mantidproject/mantid/compare/21218_refactor_GDPP_reduceRow...21457_fix_CTWA_params).

This PR fixes some bugs on the `ISIS Reflectometry` GUI to do with creating the transmission workspace and generating a python notebook:

- Fix a bug where parameters are not getting passed through to `CreateTransmissionWorkspaceAuto` from the ISIS Reflectometry GUI.
- Fix `GenerateNotebook` so that special characters are allowed in the workspace names. This is required to preserve characters such as `+` which are used in names for workspaces that have been summed.
- Fix `GenerateNotebook` so that all algorithm property values are enclosed in quotes. Previously, the notebook would not run due to some unquoted properties in `ReflectometryReductionOneAuto`.
- Make post-processing optional in `GenerateNotebook` and remove an assertion that was happening when saving a notebook from the `ISIS SANS v2 experimental` GUI.

**To test:**

Check that the bug described in #21457 is fixed.

* Open the `ISIS Reflectometry` GUI and select the `INTER` instrument.
* Expand the group and enter `13460` in the first row. Add another row with `13461+13462`.
* Tick `Output Notebook`
* On the Settings tab, enter:
  * Transmission run(s): `13463,13464`
  * dQ/Q: `0.02`
  * Stitch1DMany: `Params=-0.02`
  * IntegratedMonitors: tick yes
* Back on the Runs tab, select the group and click `Process` and select a location to save the notebook.
* [Launch](https://www.mantidproject.org/Using_IPython_Notebook) IPython Notebook and open the saved notebook. Check that it all runs ok.
* Check each of the plots in the notebook against a plot of the same workspace(s) that were generated in MantidPlot (note they are log-log plots with errors).

Fixes #21457 

**Release Notes** 
*Added to the release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
